internal/replacer/replacer_test.go

```diff
@@ -187,27 +187,81 @@ func TestReplaceWordAllCaps(t *testing.T) {
   }
 }
 
-func TestReplaceWordWithSuffix(t *testing.T) {
+func TestReplaceWordWithPrefixes(t *testing.T) {
+  t.Run("maintain prefix", func(t *testing.T) {
+    var wordmap wordmap.WordMap
+    wordmap.AddOne("-ize", "-ise")
+
+    source := "They Realize that they should not idealize."
+    result := ReplaceAll(source, wordmap)
+
+    expected := "They Realise that they should not idealise."
+    if result != expected {
+      reportIncorrectReplacement(t, expected, result)
+    }
+  })
+  t.Run("omit prefix", func(t *testing.T) {
+    var wordmap wordmap.WordMap
+    wordmap.AddOne("- people", "people")
+
+    source := "Cool people are nice and nice people are cool."
+    result := ReplaceAll(source, wordmap)
+
+    expected := "People are nice and people are cool."
+    if result != expected {
+      reportIncorrectReplacement(t, expected, result)
+    }
+  })
+}
+
+func TestReplaceWordWithSuffixes(t *testing.T) {
+  t.Run("maintain suffix", func(t *testing.T) {
+    var wordmap wordmap.WordMap
+    wordmap.AddOne("color-", "colour-")
+
+    source := "The Colors are amazing on this colorful painting. What is your favourite color?"
+    result := ReplaceAll(source, wordmap)
+
+    expected := "The Colours are amazing on this colourful painting. What is your favourite colour?"
+    if result != expected {
+      reportIncorrectReplacement(t, expected, result)
+    }
+  })
+  t.Run("omit suffix", func(t *testing.T) {
+    var wordmap wordmap.WordMap
+    wordmap.AddOne("hack-", "hackers")
+
+    source := "Some hackz0rs are good while some hacknerds are evil."
```

Comment:
Replace this with [the example in the documentation](https://github.com/ericcornelissen/wordrow/blob/feat-20/suffix-and-prefix-support/docs/mapping-files.md#omitting-prefixes-or-suffixes)

```suggestion
    source := "I have a dog, but you have a small doggy."
```